### PR TITLE
[JS] Topic: Functions

### LINKED ISSUE #247: JS: Functions
# Functions
Functions in JavaScript

## Directory
```
Topics/5.Syntax/6.Functions
```

## Description
- What is scope?
- What is the lifetime of JavaScript variables?
- Basic function syntax.
- What are parameters in functions?
- Types of functions.
- Why use functions?

### PR DESCRIPTION
**Pull request by siddhi-244**
<hr>

## Description 📜

Added brief into about functions in javascript. 
Fixes #247 

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ 

- [x] JavaScript

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by Creatoon** [comment]
@siddhi-244 Also check for grammatical errors.

**Comment by siddhi-244** [comment]
> @siddhi-244 Also check for grammatical errors.

Sure

**Comment by siddhi-244** [comment]
Done !

File: Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md [hunk 1]
```diff
@@ -90,7 +90,3 @@ limited and can't be used in all situations.
 ## Why use functions?
 Functions are a good alternative to having repeating blocks of code in a program. Functions also increase the reusability of code. Values can be passed to a function using variables – we call these parameters or arguments.
 Functions can also return values.
```
Comment (Creatoon) [inline Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md:92]:
```suggestion
Functions can also return values.

```

**Review by Creatoon — CHANGES_REQUESTED** [review]

File: Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md (hunk 1 above)
Comment (Creatoon) [reply Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md:92]: I said leave one line, just commit it

**Comment by siddhi-244** [comment]
Done

**Comment by siddhi-244** [comment]
> @siddhi-244 Also check for grammatical errors.

Sorry I have done all the changes

**Comment by paramsiddharth** [comment]
@siddhi-244 You seem to have pulled updates from `girlscript:main` to the branch for this PR, whereas you should have pulled from `girlscript:Javascript` to this branch. :) Although that doesn't cause any errors here, be careful as this may introduce inconsistensies.

Since your PR is the first one for #247, you are at priority. Only one mentor is active so I'll not wait for further reviews. I'll review it and merge it as soon as possible.

**Comment by siddhi-244** [comment]
> @siddhi-244 You seem to have pulled updates from `girlscript:main` to the branch for this PR, whereas you should have pulled from `girlscript:Javascript` to this branch. :) Although that doesn't cause any errors here, be careful as this may introduce inconsistensies.
> 
> Since your PR is the first one for #247, you are at priority. Only one mentor is active so I'll not wait for further reviews. I'll review it and merge it as soon as possible.

I am really srry. I will see to it next time.

File: Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+(a) => {
+  return a + 100;
+}
+```
+<hr>
+
+## Why use functions?
+Functions are a good alternative to having repeating blocks of code in a program. Functions also increase the reusability of code. Values can be passed to a function using variables – we call these parameters or arguments.
+Functions can also return values.
+
```
Comment (paramsiddharth) [inline Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md:93]: You might want to mention your name here as a contributor.
```
Contributor: Siddhi
```
Please update the PR.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Comment by siddhi-244** [comment]
Done @paramsiddharth sir

**Review by paramsiddharth — APPROVED** [review]

**Review by Creatoon — APPROVED** [review]

**Review by Creatoon — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@dakshsethi Could you please review the PR, and request any changes if needed?

File: Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+var a=100 ;
+console.log(a); // 100
+```
+You can easily console log this variable. Let's consider another 
+example
+``` javascript 
+if (true) {
+  const a = 2;
+}
+console.log(a); //ReferenceError
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/6.Functions/README.md:13]: Mention the error message exactly as it shows in the console window `ReferenceError: a is undefined`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

**Comment by siddhi-244** [comment]
Done Sir @dakshsethi

**Comment by paramsiddharth** [comment]
@dakshsethi Please review further. :)

**Review by dakshsethi — APPROVED** [review]
Looks good to me!!

**Pull request merged**